tensorflow/compiler/tf2tensorrt/convert/utils.h

```diff
@@ -137,6 +138,26 @@ inline bool IsTrtShapeTensorCompatible(const Tensor& tensor) {
          IsTrtShapeTensorCompatible(tensor.shape());
 }
 
+template <typename Container>
+Status ContainerToTrtDims(const Container& shape, nvinfer1::Dims* trt_dims,
+                          bool ignore_first_dim = false) {
+  if (shape.size() == 0) {
+    // scalar
+    if (ignore_first_dim) {
+      return errors::Internal(
+          "Scalars cannot be represented in implicit batch mode");
+    }
+    *trt_dims = {0, {1}};
+  } else {
+    const int offset = (ignore_first_dim ? 1 : 0);
+    for (int i = offset; i < shape.size(); i++) {
+      trt_dims->d[i - offset] = shape.at(i);
+    }
+    trt_dims->nbDims = shape.size() - offset;
+  }
+  return Status::OK();
+}
+
 template <typename TensorShapeType>
 Status TensorShapeToTrtDims(const TensorShapeType& shape, bool ignore_first_dim,
                             nvinfer1::Dims* trt_dims) {
```

Comment:
Let's replace this routine body with this:
  if (shape.dims() == -1) {
    trt_dims->nbDims = -1;
    return Status::OK();
  }
  return ContainerToTrtDims(shape.dim_sizes(), trt_dims, ignore_first_dim);

Reply:
Thanks for the suggestion, I have changed it.